web/web/files--common/javascript/OZONE/dialog.ts

```diff
@@ -0,0 +1,675 @@
+import CSS from "csstype";
+
+import OZONE from ".";
+
+declare const YAHOO: any;
+declare const fx: any;
+
+/**
+ * Backend for dialog boxes and hovertips.
+ *
+ * The shader (#odialog-shader )is the dark bit around the dialog box that
+ * obscures the rest of the page.
+ * The box container (#odialog-container) contains the dialog box (.owindow).
+ */
+
+// The hovertips code features elements with hovertip-related properties
+// attached.
+type HovertipOptions = {
+  context?: string
+  text?: string
+  style?: CSS.Properties
+  noCursorHelp?: boolean
+  delay?: number
+  smartWidthLimit?: number
+  valign?: 'center'
+}
```

Comment:
Related to the point about `interface`, shouldn't we end each of these in `;`?

Reply:
With ESLint set to be uncompromising on semicolons, it doesn't complain about this. The single-line form of an object type declaration is delimited by commas, but ESLint would normally complain about that too and it doesn't here. Maybe because this never makes it to Javascript?